src/parquet/column/scanner.h

```diff
@@ -45,7 +45,7 @@ class Scanner {
       values_buffered_(0),
       reader_(reader) {
     // TODO: don't allocate for required fields
-    def_levels_.resize(reader->descr()->is_optional() ? batch_size_ : 0);
+    def_levels_.resize(!reader->descr()->is_required() ? batch_size_ : 0);
```

Comment:
Infact, when I changed the implementation to return the `RepetitionType` in the `PrimitiveNode`, the previous code breaks even for flat schema.